helix-core/src/main/java/org/apache/helix/HelixAdmin.java

```diff
@@ -228,6 +228,13 @@ void addResource(String clusterName, String resourceName, int numPartitions, Str
    */
   void dropInstance(String clusterName, InstanceConfig instanceConfig);
 
+  /**
+   * Purge offline instances from a cluster
+   * @param clusterName
+   * @param timeout
+   */
+  void purgeOfflineInstances(String clusterName, Long timeout);
```

Comment:
i think the javadoc here could use more work. we should try to add more context on why and when the interface should be used. 

also i'm not sure `timeout` is the right term for this. perhaps rename it to something like `offlineDuration`? also if you could please add a description as to how this `offlineDuration` is measured/calculated.

Reply:
also is there a strong reason why we don't use a primitive `long`?